Algorithm.CSharp/Alphas/PumpAndDumpAlpha.cs

```diff
@@ -0,0 +1,129 @@
+    /// You can read the source code for this alpha on Github in <a href="https://github.com/QuantConnect/Lean/blob/master/Algorithm.CSharp/PumpAndDumpAlpha.cs">C#</a>
+    /// or <a href="https://github.com/QuantConnect/Lean/blob/master/Algorithm.Python/Alphas/PumpAndDumpAlpha.py">Python</a>.
+    /// </summary>
+    public class PumpAndDumpAlpha : QCAlgorithmFramework
+    {
+        public override void Initialize()
+        {
+            SetStartDate(2018, 1, 1);
+            SetCash(100000);
+
+            // Set zero transaction fees
+            SetSecurityInitializer(security => security.FeeModel = new ConstantFeeModel(0));
+
+            // Select stocks using PennyStockUniverseSelectionModel
+            UniverseSettings.Resolution = Resolution.Daily;
+            SetUniverseSelection(new PennyStockUniverseSelectionModel());
+
+            // Use PumpAndDumpAlphaModel to establish insights
+            SetAlpha(new PumpAndDumpAlphaModel());
+
+            // Equally weigh securities in portfolio, based on insights
+            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
+        }
+
+        /// <summary>
+        /// Performs coarse selection for constituents.
+        /// The stocks must have fundamental data
+        /// The stock must have positive previous-day close price
+        /// The stock must have volume between $1000000 and $10000 on the previous trading day
+        /// The stock must cost less than $5'''
+        /// </summary>
+        private class PennyStockUniverseSelectionModel : FundamentalUniverseSelectionModel
+        {
+            private const int _numberOfSymbolsCoarse = 500;
+            private int _lastMonth = -1;
+
+            public PennyStockUniverseSelectionModel() : base(true, null, null)
+            {
+            }
+
+            public override IEnumerable<Symbol> SelectCoarse(QCAlgorithmFramework algorithm, IEnumerable<CoarseFundamental> coarse)
+            {
+                var month = algorithm.Time.Month;
+                if (month == _lastMonth)
+                {
+                    return algorithm.Universe.Unchanged;
+                }
+                _lastMonth = month;
+
+                return (from cf in coarse
+                        where cf.HasFundamentalData
+                        where cf.Volume < 1000000
+                        where cf.Volume > 10000
+                        where cf.Price < 5
+                        orderby cf.DollarVolume descending
+                        select cf.Symbol).Take(_numberOfSymbolsCoarse);
+            }
+        }
+
+        /// <summary>
+        /// Uses ranking of intraday percentage difference between open price and close price to create magnitude and direction prediction for insights
+        /// </summary>
+        private class PumpAndDumpAlphaModel : AlphaModel
```

Comment:
Please rename this alpha - and the python version to: "SykesShortMicroCapAlphaModel" -> and the algorithm to: "SykesShortMicroCapAlpha"